packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts

```diff
@@ -392,3 +400,74 @@ export const onlyAccountWithMinimumBalance = ({
     );
   }
 };
+
+// ---------------------Voucher checks------------------------------------
+export const onlyVoucherAuthorizingSharingContract = async ({
+  sharingContractAddress,
+  voucherContract,
+}: {
+  sharingContractAddress: Address;
+  voucherContract: IVoucher;
+}) => {
+  const [voucherContractAddress, isAuthorizedToUseVoucher] = await Promise.all([
+    voucherContract.getAddress(),
+    voucherContract.isAccountAuthorized(sharingContractAddress),
+  ]);
+
+  if (!isAuthorizedToUseVoucher) {
+    throw new Error(
+      `The sharing contract (${sharingContractAddress}) is not authorized to use the voucher ${voucherContractAddress}. Please authorize it to use the voucher.`
+    );
+  }
+};
+
+export const onlyFullySponsorableAssets = async ({
+  pocoContract,
+  voucherHubContract,
+  voucherContract,
+  userAddress,
+  workerpoolOrder,
+}: {
+  pocoContract: IExecPocoDelegate;
+  voucherHubContract: IVoucherHub;
+  voucherContract: IVoucher;
+  userAddress: Address;
+  workerpoolOrder: any; // TODO: Find the corresponding type
+}) => {
+  const voucherContractAddress = await voucherContract.getAddress();
+
+  const [isWorkerpoolSponsoredByVoucher, voucherBalance, accountAllowance] =
+    await Promise.all([
+      isAnEligibleAsset({
+        voucherHubContract,
+        voucherContract,
+        assetAddress: workerpoolOrder.workerpool,
+      }),
+      getVoucherBalance({ voucherContract }),
+      getAccountAllowance({
+        pocoContract,
+        owner: userAddress,
+        spender: voucherContractAddress,
+      }),
+    ]);
+
+  if (!isWorkerpoolSponsoredByVoucher) {
+    throw new Error(
+      `${workerpoolOrder.workerpool} is not sponsored by the voucher ${voucherContractAddress}`
+    );
+  }
+
+  const workerpoolPrice = Number(workerpoolOrder.workerpoolprice);
+  if (voucherBalance < workerpoolPrice) {
+    const missingAmount = workerpoolPrice - Number(voucherBalance);
+
+    if (
+      Number(accountAllowance) === 0 ||
+      Number(accountAllowance) < workerpoolPrice
+    ) {
+      throw new Error(
+        `Voucher balance is insufficient to sponsor workerpool. Please approve an additional ${missingAmount} for voucher usage.`
+      );
+    }
+  }
```

Comment:
do the maths with bigints instead of casting to number

Reply:
fix :Commit: 7c16db7d76f4fa4003163f82e720ef98c9a9a5a3